api/buckets/buckets.go

```diff
@@ -106,23 +182,258 @@ func (b *Buckets) Save(ctx context.Context, dbID thread.ID, bucket *Bucket, opts
 	for _, opt := range opts {
 		opt(args)
 	}
-	return b.Threads.Save(ctx, dbID, CollectionName, dbc.Instances{bucket}, db.WithTxnToken(args.Token))
+	return b.threads.Save(ctx, dbID, CollectionName, dbc.Instances{bucket}, db.WithTxnToken(args.Token))
 }
 
 func (b *Buckets) Delete(ctx context.Context, dbID thread.ID, key string, opts ...Option) error {
 	args := &Options{}
 	for _, opt := range opts {
 		opt(args)
 	}
-	return b.Threads.Delete(ctx, dbID, CollectionName, []string{key}, db.WithTxnToken(args.Token))
+	return b.threads.Delete(ctx, dbID, CollectionName, []string{key}, db.WithTxnToken(args.Token))
+}
+
+// Archive pushes the current root Cid to the corresponding FFS instance of the bucket.
+// If it's not the first archive, we leverage the Replace() API to more efficient pinning.
+// The operation is async, so it will return as soon as the archiving is scheduled in Powergate.
+func (b *Buckets) Archive(ctx context.Context, dbID thread.ID, key string, c cid.Cid, opts ...Option) error {
+	b.lock.Lock()
+	defer b.lock.Unlock()
+	ffsi, err := b.ffsCol.Get(ctx, key)
+	if err != nil {
+		return fmt.Errorf("getting ffs instance data: %s", err)
+	}
+
+	ctxFFS := context.WithValue(ctx, powc.AuthKey, ffsi.FFSToken)
+	var jid ffs.JobID
+	if ffsi.Archives.Current.JobID != "" {
```

Comment:
We make here the distinction to call `PushConfig` if it's the first archive, or `Replace` if its a further one.